test/e2e/kappcontroller/installation_namespace_test.go

```diff
@@ -0,0 +1,288 @@
+
+	installPkgYAML := fmt.Sprintf(`
+          ---
+          apiVersion: data.packaging.carvel.dev/v1alpha1
+          kind: PackageMetadata
+          metadata:
+            name: pkg.test.carvel.dev
+            namespace: %[1]s
+          spec:
+            # This is the name we want to reference in resources such as PackageInstall.
+            displayName: "Test PackageMetadata in repo"
+            shortDescription: "PackageMetadata used for testing"
+            longDescription: "A longer, more detailed description of what the package contains and what it is for"
+            providerName: Carvel
+            maintainers:
+            - name: carvel
+            categories:
+            - testing
+            supportDescription: "Description of support provided for the package"
+          ---
+          apiVersion: data.packaging.carvel.dev/v1alpha1
+          kind: Package
+          metadata:
+            name: pkg.test.carvel.dev.1.0.0
+            namespace: %[1]s
+          spec:
+            refName: pkg.test.carvel.dev
+            version: 1.0.0
+            licenses:
+            - Apache 2.0
+            capactiyRequirementsDescription: "cpu: 1,RAM: 2, Disk: 3"
+            releaseNotes: |
+              - Introduce simple-app package
+            releasedAt: 2021-05-05T18:57:06Z
+            template:
+              spec:
+                fetch:
+                - inline:
+                    paths:
+                      file.yml: |
+                        apiVersion: v1
+                        kind: ConfigMap
+                        metadata:
+                          name: my-cm
+                        data:
+                          key: value
+                template:
+                - ytt: {}
+                deploy:
+                - kapp:
+                    rawOptions: ["--app-changes-max-to-keep=0"]
+          ---
+          apiVersion: packaging.carvel.dev/v1alpha1
+          kind: PackageInstall
+          metadata:
+            name: %[2]s
+            namespace: %[1]s
+            annotations:
+          spec:
+            installationNamespace: %[3]s
+            serviceAccountName: kappctrl-e2e-ns-sa
+            packageRef:
+              refName: pkg.test.carvel.dev
+              versionSelection:
+                constraints: 1.0.0
+`, pkgiNamespace, pkgiName, installationNamespace)
+
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	// deploy app with workspace and pkgi
+	_, err := kapp.RunWithOpts([]string{"deploy", "-a", name, "-f", "-"}, e2e.RunOpts{StdinReader: strings.NewReader(appWithInlineResources(appName, namespacesYAML+rbacYAML+installPkgYAML) + sas.ForClusterYAML())})
+	if err != nil {
+		t.Fatalf("Expected app deploy to succeed, it did not: %v", err)
+	}
+
+	// assert that pkgi resources are present in installation namespace
+	_, err = kubectl.RunWithOpts([]string{"get", "configmap", "my-cm", "-n", installationNamespace}, e2e.RunOpts{AllowError: true, NoNamespace: true})
```

Comment:
```suggestion
	_, err = kubectl.RunWithOpts([]string{"get", "configmap", "my-cm", "-n", installationNamespace}, e2e.RunOpts{AllowError: false, NoNamespace: true})
```

Should this be false on the error allowed? Same as one below

Reply:
I typically prefer not using AllowError in these scenarios so that the kubectl command fails, but I copied this style from other tests where we allow the error to occur and then wrap the error in a message.
I am okay with setting allow error to false.